src/lib/encoder.ts

```diff
@@ -313,20 +315,32 @@ const decodeStringCompactBytesArray = (compactBytesArray: string): string[] => {
 
 const valueTypeEncodingMap = {
   bool: {
-    encode: (value: boolean) => abiCoder.encodeParameter('bool', value),
-    decode: (value: string) => abiCoder.decodeParameter('bool', value),
+    encode: (value: boolean) => (value ? '0x01' : '0x00'),
+    decode: (value: string) => value === '0x01',
   },
   boolean: {
-    encode: (value: boolean) => abiCoder.encodeParameter('bool', value),
-    decode: (value: string) => abiCoder.decodeParameter('bool', value),
+    encode: (value: boolean) => (value ? '0x01' : '0x00'),
+    decode: (value: string) => value === '0x01',
   },
   string: {
-    encode: (value: string) => abiCoder.encodeParameter('string', value),
-    decode: (value: string) => abiCoder.decodeParameter('string', value),
+    encode: (value: string | number) => {
+      // if we receive a number as input,
+      // convert each letter to its utf8 hex representation
+      if (typeof value === 'number') {
+        return utf8ToHex(`${value}`);
+      }
+
+      return utf8ToHex(value);
+    },
+    decode: (value: string) => hexToUtf8(value),
   },
   address: {
-    encode: (value: string) => abiCoder.encodeParameter('address', value),
-    decode: (value: string) => abiCoder.decodeParameter('address', value),
+    encode: (value: string) => {
+      const abiEncodedValue = abiCoder.encodeParameter('address', value);
+      const bytesArray = hexToBytes(abiEncodedValue);
+      return bytesToHex(bytesArray.slice(12));
```

Comment:
why do you need the last 2 lines?

Why do you do hexToBytes then bytesToHex on sliced version?

The abicoder is not enough?

Maybe would be nice to add a comment here to explain why you have to do this. This confuses me.

Reply:
Here I did `hexToBytes` and then `bytesToHex` because it is easier to reason in `bytes` in the library.

Otherwise I could do the following screenshot below, but I believe it is much harder to reas and understand because you have to _"calculate in your head"_.

```
0x = 2 characters
12 first bytes = 24 characters

2 + 24 = 26
```

Very hard to understand, I don't like it at all.